pilot/pkg/xds/auth.go

```diff
@@ -57,15 +58,18 @@ func (s *DiscoveryServer) authenticate(ctx context.Context) ([]string, error) {
 		return nil, nil
 	}
 	authFailMsgs := []string{}
+	authContext := security.NewAuthContext(ctx)
+
 	for _, authn := range s.Authenticators {
-		u, err := authn.Authenticate(ctx)
+		authContext.AddAuthenticator(authn.AuthenticatorType())
+		u, err := authn.Authenticate(authContext)
 		// If one authenticator passes, return
 		if u != nil && u.Identities != nil && err == nil {
+			authContext.Caller = u
```

Comment:
The problem with 'identities' is that there is no guarantee there is only one identity and it is spiffee first. 
The request may have multiple identities - but we care only about the spiffee one ( and really about namespace + KSA - Istiod is not concerned with spiffee, just namespace ).